### PR TITLE
feat: respect `RATTLER_AUTH_FILE` when using AuthenticationStorage::default()

### PR DESCRIPTION
**Pull request by 0xbe7a**
The environment variable `RATTLER_AUTH_FILE` is currently used by both `pixi` and `rattler-build` to specify an alternative storage location for the `credentials.json` file, which is crucial for authenticating to remote repositories. This feature is particularly valuable in CI or managed environments, where authentication is managed externally and credentials are stored at ephemeral paths.

Currently, both tools handle the `RATTLER_AUTH_FILE` independently, each tool implementing its own method to default to `AuthenticationStorage` when the environment variable is not set. This is evidenced in their respective codebases:

- `rattler-build` implementation: [rattler-build code](https://github.com/prefix-dev/rattler-build/blob/main/src/tool_configuration.rs#L57)
- `pixi` implementation: [pixi code](https://github.com/prefix-dev/pixi/blob/main/src/utils/reqwest.rs#L22)

Despite their similar approaches, this independent handling has led to at least one consistency issue as seen in [pixi issue #1240](https://github.com/prefix-dev/pixi/issues/1240).

To address this and enhance consistency across tools that utilize `RATTLER_AUTH_FILE`, I propose integrating the handling of this environment variable directly into the default implementation (`impl Default`) of `AuthenticationStorage`. This change will ensure that any tool relying on `AuthenticationStorage` will inherently respect the `RATTLER_AUTH_FILE` path if set, without requiring each tool to implement its own handling. The proposed change aligns with the intended use of the environment variable, suggesting a default behavior that could be expected across all tools built on the rattler framework.

This modification will not remove flexibility, as developers can still explicitly construct `AuthenticationStorage` instances without this default behavior if needed.

### CODE REVIEW

**Review by baszalmstra — CHANGES_REQUESTED** [review]
I apologize for leaving this PR for so long! It completely missed my radar!

I am strongly opposed to reading environment variables in normal happy flow code. So I would prefer to move this implementation to a function that is explicit about reading this from the environment. I added a suggestion as a review comment.

File: crates/rattler_networking/src/authentication_storage/storage.rs [hunk 1]
```diff
@@ -29,6 +29,23 @@ impl Default for AuthenticationStorage {
...
+            );
+
+            let backend = FileStorage::new(path.to_path_buf()).unwrap_or_else(|e| {
+                tracing::error!("Failed to create file storage at {}: {}", auth_file, e);
+                FileStorage::default()
+            });
+
+            storage.add_backend(Arc::from(backend));
+            return storage;
+        }
```
[the remark this answers is not in the document]
Comment (0xbe7a) [reply crates/rattler_networking/src/authentication_storage/storage.rs]: I added `AuthenticationStorage::from_env `

File: crates/rattler_networking/src/authentication_storage/storage.rs [hunk 2]
```diff
@@ -51,6 +51,33 @@ impl AuthenticationStorage {
         }
     }
 
+    /// Create a new authentication storage with the default backends
+    /// respecting the `RATTLER_AUTH_FILE` environment variable
```
Comment (baszalmstra) [inline crates/rattler_networking/src/authentication_storage/storage.rs:55]: Maybe you can explain what "respecting the `RATTLER_AUTH_FILE`" means?

**Review by baszalmstra — APPROVED** [review]

**Comment by baszalmstra** [comment]
Thanks!

**Pull request merged**